src/material/form-field/form-field.ts

```diff
@@ -365,17 +368,23 @@ export class MatFormField
 
   ngAfterContentInit() {
     this._assertFormFieldControl();
-    this._initializeControl();
     this._initializeSubscript();
     this._initializePrefixAndSuffix();
     this._initializeOutlineLabelOffsetSubscriptions();
   }
 
   ngAfterContentChecked() {
     this._assertFormFieldControl();
+
+    if (this._control !== this._previousControl) {
```

Comment:
Another option for this would've been to use signals, but it seems like `_control` is set up as some sort of backwards compatibility layer so I didn't want to change it into a computed which would've been breaking.